### ISSUE TITLE
Scaffold for Linux desktop

### ISSUE THREAD

**Issue by hpoul** [authpass/authpass#26]

**Comment by hpoul** [comment]
fwiw, the desktop linux release mode (with the glfw, not gtk) has been merged into flutter master 🎉️
i've tried a basic `flutter create` and it indeed looks promising.. i guess i'll play around the next few days to get the bare minimum of plugins working.. (i could imagine argon2 (kdbx 4) support to be the most troublesome..) we'll see..

<img width="1070" alt="Screenshot 2020-05-30 at 14 14 38" src="https://user-images.githubusercontent.com/313066/83327932-e97c3880-a27f-11ea-8270-556b6f5f4532.png">

**Comment by hpoul** [comment]
I have now a reasonable first binary in release mode ready which supports basic functions (like opening kdbx files :) and also argon2 for kdbx 4..) Anyone willing to give it a try?

[authpass-linux-1186.tar.gz](https://github.com/authpass/authpass/files/4718016/authpass-linux-1186.tar.gz)

**Comment by idunno101** [comment]
Just tested it on Fedora 32 (Wayland).
It worked well, I opened a local database (Argon2 KDBX 4) and was able to filter entries, copy values and add a new entry.
There were a few errors registered in the terminal.

This error showed up when I first launched the program but didn't show up on any launch after that.
[Link 1](https://bin.privacytools.io/?532f6b477f4502ab#Gtc862O2uTVfKjaniCjZCnEsDehvALLL8onhMInlbDA=)

These two were triggered when the database was unlocked
[Link 2](https://bin.privacytools.io/?7011822de8a5216b#nRBnyZr7hdkvHFFxvMD3yWlATlpTsCLRtAfe7nw5jWk=)
[Link 3](https://bin.privacytools.io/?a74d827e1b21e20b#6Pj/izJKKUdONAqRUBYdoS75QH2BLnsHIKGsnkCBffI=)

This one was triggered when the database was locked
[Link 4](https://bin.privacytools.io/?84404cd2207606cd#y4jZgTNQnqnSIIuev/XjeTloM6QAXG8EVcrSZQvPSGM=)

And this one happened when viewing the entries that didn't have the otp set up as 
otpauth://totp/Site?secret&time
I have some logins (like Google and Firefox) that set their TOTP as two separate entries; one is a seed (like XXXX XXXX) and the other is a time url (like 30;6;https://time.is/)
For those logins it gave this error and didn't display the otp digits
[Link 5](https://bin.privacytools.io/?79ca58ce4d7644fc#vmVDSyWMwoU5gVwa0TuanZpFZzV5ut7sVFkJ+m1LQbI=)

**Comment by hpoul** [comment]
@idunno101 thanks for the exhaustive test 👍️ i have also seen the first error (regarding clipboard) occassionally.. can you still copy password (ctrl+c) and usernames (ctrl+b)? (or via the dropdown button next to the fields)

the last error regarding TOTP shouldn't be related to the linux version obviously.. how did you create those entries? would it be easy for you to create an empty kdbx file with dummy entries in the same format, so I can reproduce it myself?

**Comment by idunno101** [comment]
> can you still copy password (ctrl+c) and usernames (ctrl+b)? (or via the dropdown button next to the fields)

Yeah both methods worked even while the errors were being triggered.

> how did you create those entries? would it be easy for you to create an empty kdbx file with dummy entries in the same format, so I can reproduce it myself?

I looked through my logins a bit more and it seems that the problem occurs when there are spaces within the seed. I was able to reproduce the behaviour in the database attached here 
Database password is 123
[totp test.zip](https://github.com/authpass/authpass/files/4726932/totp.test.zip)

**Comment by hpoul** [comment]
@idunno101 thanks for the test database. You are exactly right, the spaces broke everything.. (I clean up those keys when the user adds the TOTP secrets..). Anyway, it is now fixed and I've used your test data to [create unit tests. So this should stay fixed](https://github.com/authpass/authpass/blob/9cd1718983da682084d56393cdd3687d74df4091/authpass/test/entry_details_test.dart#L69-L100 :)

If you like to try again, here is an updated linux version - it should also fix a couple of other exceptions (unfortunately not the one regarding clipboard, because that's a problem in the flutter embedder):

[authpass-linux-1.6.3_1196.zip](https://github.com/authpass/authpass/files/4729086/authpass-linux-1.6.3_1196.zip)

**Comment by idunno101** [comment]
Yep looks good :smiley:   Didn't notice any problems
